app/src/main/java/mod/RockAriful/AndroXStudio/GithubConfigActivity.java

```diff
@@ -0,0 +1,474 @@
+import org.eclipse.jgit.api.errors.*;
+import java.io.BufferedWriter; 
+import java.io.File; 
+import java.io.FileOutputStream; 
+import java.io.OutputStreamWriter; 
+import java.text.MessageFormat; 
+import java.util.Date; 
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean; 
+ 
+import org.eclipse.jgit.api.CloneCommand; 
+import org.eclipse.jgit.api.Git; 
+import org.eclipse.jgit.api.ResetCommand.ResetType; 
+import org.eclipse.jgit.api.ListBranchCommand.ListMode;
+import org.eclipse.jgit.api.errors.GitAPIException; 
+import org.eclipse.jgit.lib.Constants; 
+import org.eclipse.jgit.revwalk.RevCommit; 
+import org.eclipse.jgit.transport.CredentialsProvider; 
+import org.eclipse.jgit.transport.PushResult; 
+import org.eclipse.jgit.transport.RefSpec; 
+import org.eclipse.jgit.transport.*;
+import org.eclipse.jgit.transport.RemoteRefUpdate; 
+import org.eclipse.jgit.transport.RemoteRefUpdate.Status; 
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider; 
+import org.eclipse.jgit.lib.*;
+import org.eclipse.jgit.util.FileUtils; 
+
+public class GithubConfigActivity extends AppCompatActivity {
+	
+	private Toolbar _toolbar;
+	private AppBarLayout _app_bar;
+	private CoordinatorLayout _coordinator;
+	private HashMap<String, Object> map = new HashMap<>();
+	
+	private ArrayList<HashMap<String, Object>> JsonMAP = new ArrayList<>();
+	
+	private ScrollView vscroll1;
+	private LinearLayout linear1;
+	private LinearLayout linear2;
+	private LinearLayout linear13;
+	private LinearLayout all_item_layout;
+	private LinearLayout linear14;
+    
+   	 private LinearLayout push_data;
+        private ProgressBar progressbar1;
+        private TextView push_btn_title;
+	private TextView create_token;
+    
+	private TextView textview1;
+	private Switch enable;
+	private ImageView imageview2;
+	private TextView textview13;
+	private LinearLayout is_url;
+	private LinearLayout linear12;
+	private EditText setRefSpecs;
+	private LinearLayout credential_layout;
+	private TextView textview7;
+	private EditText url;
+	private TextView textview11;
+	private TextView important_btn;
+	private TextView textview5;
+	private EditText username;
+	private TextView textview6;
+	private EditText pass;
+	private LinearLayout radio_group;
+    private String exportedSourcesPath ="";
+    private Intent Ctoken = new Intent();
+    private Intent GoToLog = new Intent();
+    
+    private String sc_id ="";
+    
+
+	//boolean&string for fatchin fushing result
+	private boolean isSucces = false;
+    
+    private Calendar calender = Calendar.getInstance();
+	
+	@Override
+	protected void onCreate(Bundle _savedInstanceState) {
+		super.onCreate(_savedInstanceState);
+		setContentView(R.layout.github_config);
+		initialize(_savedInstanceState);
+		_toolbar.setPopupTheme(R.style.ThemeOverlay_ToolbarMenu);
+		if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_DENIED

```

Comment:
These permissions should already be granted by the time the user comes here, could be removed